### DISCUSSION TITLE
Storing results on the codechecker UI

### DISCUSSION THREAD

**Discussion by RakshanPremsagar** [Ericsson/codechecker#4240] [Q&A]
I have generated the compilation database(json file) and have analysed them to a results folder( This process is on a different container - Lets say A)

 I have the codechecker container (server) running at port 8001. I want to store the results from my anaylsis on to the server.

After the anaylsis was done, I wrote the following command on the container A: 
`CodeChecker store reports --name example --url http://codechecker.de/Default`

I get the following response: 
```

[INFO 2024-05-08 06:37] - Storing analysis results for run 'example'
[INFO 2024-05-08 06:37] - Checking local passwords or tokens in /root/.codechecker.passwords.json
[INFO 2024-05-08 06:37] - Password file not found.
[INFO 2024-05-08 06:37] - Checking for local valid sessions.
[ERROR 2024-05-08 06:37] - Thrift invalid data error.
[ERROR 2024-05-08 06:37] - getAuthParameters
[ERROR 2024-05-08 06:37] - []
[ERROR 2024-05-08 06:37] - {}
[ERROR 2024-05-08 06:37] - Request failed.
Traceback (most recent call last):
  File "/codechecker/build/CodeChecker/lib/python3/codechecker_client/thrift_call.py", line 45, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/codechecker_api/Authentication_v6/codeCheckerAuthentication.py", line 162, in getAuthParameters
    return self.recv_getAuthParameters()
  File "/usr/local/lib/python3.8/dist-packages/codechecker_api/Authentication_v6/codeCheckerAuthentication.py", line 173, in recv_getAuthParameters
    (fname, mtype, rseqid) = iprot.readMessageBegin()
  File "/usr/local/lib/python3.8/dist-packages/thrift/protocol/TJSONProtocol.py", line 417, in readMessageBegin
    self.readJSONArrayStart()
  File "/usr/local/lib/python3.8/dist-packages/thrift/protocol/TJSONProtocol.py", line 405, in readJSONArrayStart
    self.readJSONSyntaxChar(LBRACKET)
  File "/usr/local/lib/python3.8/dist-packages/thrift/protocol/TJSONProtocol.py", line 255, in readJSONSyntaxChar
    raise TProtocolException(TProtocolException.INVALID_DATA,
thrift.protocol.TProtocol.TProtocolException: Unexpected character: b'<'
```

Questions: 

1. Is it the case that the store method needs to be run on the CodeChecker container?
2. When setting up authentication for the server, we modify the server_config, add the users there and expect the new users to login on container restart?
3. What is with the `[INFO 2024-05-08 06:37] - Checking local passwords or tokens in /root/.codechecker.passwords.json`? Do we have to setup another file here?

**Comment by bruntib** [comment]
Hi @RakshanPremsagar,

1. The CodeChecker server should be available from the machine where `CodeChecker store` runs. Make sure that the network connection between them is properly set. By default, CodeChecker enables local connections only. Try starting the server like this: `CodeChecker server --not-host-only`. Also, provide the port number at storage, otherwise the default port 80 is used: `CodeChecker store --url https://codechecker.de:8001/Default`. It might be useful to know that the default value of `--url` flag is `http://localhost:8001/Default`.
2. Yes, if you setup the authentication in server config, then server needs to restart to read this information.
3. If you setup authentication to the server, then users need to login first: `CodeChecker cmd login`. The file `.codechecker.passwords.json` shouldn't be written manually.

The error message above indicates a network issue, so I would say that the answer should be around question 1.
